spec/commands/binstubs_spec.rb

```diff
@@ -124,13 +124,51 @@
         bundle! "binstubs bundler rack prints_loaded_gems"
       end
 
-      let(:system_bundler_version) { Bundler::VERSION }
+      context "when system bundler was used" do
+        # When environment has a same version of bundler as default gems.
+        # `system_gems "bundler-x.y.z"` will detect system binstub.
+        # We need to avoid it by virtual version of bundler.
+        let(:system_bundler_version) { Gem::Version.new(Bundler::VERSION).bump.to_s }
+
+        # Support master branch of bundler
+        if ENV["BUNDLER_SPEC_SUB_VERSION"]
+          let(:system_bundler_version) { Bundler::VERSION }
+        end
+
+        before do
+          gemfile <<-G
+            source "https://rubygems.org"
+            gem "rack"
+            gem "prints_loaded_gems"
+          G
+
+          lockfile <<-G
+          GEM
+            remote: https://rubygems.org
+            specs:
+              prints_loaded_gems (1.0)
+              rack (1.2)
+
+          PLATFORMS
+            ruby
+
+          DEPENDENCIES
+            prints_loaded_gems
+            rack
+
+          BUNDLED WITH
+             #{system_bundler_version}
+          G
+        end
 
-      it "runs bundler" do
-        sys_exec! "#{bundled_app("bin/bundle")} install"
-        expect(out).to eq %(system bundler #{system_bundler_version}\n["install"])
+        it "runs bundler" do
+          sys_exec! "#{bundled_app("bin/bundle")} install"
+          expect(out).to eq %(system bundler #{system_bundler_version}\n["install"])
+        end
       end
 
+      let(:system_bundler_version) { Bundler::VERSION }
```

Comment:
Is this let necessary or the product of a bad conflict resolution? Just asking because there's 3 `let`s defining the same variable in very similar ways.

Reply:
Still wondering about this, I'll have a closer look tomorrow.

Reply:
It's required but the merge was a bit weird i think as well. I've made a new commit to fix this up.